### PR TITLE
feat: Provided ability to disable instrumentation for a 3rd party package

### LINKED ISSUE #2536: Provide ability to disable instrumentation for a given 3rd party library
### Description
We've had requests over the years to disable instrumentation for a given 3rd party library.  We were hesitant when we still had the legacy context manager due to possible loss of async context.  Now that the legacy context manager has been removed, we should allow this feature.

### Acceptance Criteria
As a Node.js customer I want the ability to disable instrumentation for a given 3rd party library so that I can work around possible bugs, favor custom instrumentation or not record telemetry for a certain library.

### Additional context
We should consult with other languages on their configuration stanza to see if there's a consistent configuration.  We need to write some documentation around this and explain the risks.  Specifically for libraries that create the transactions. If you disable those libraries no other telemetry will probably function.  Some things to test:

 * Disable webframeworks as they are responsible for transaction naming, verify 3rd party l libraries within the middleware are still recording spans/metrics/etc
 *  Disable other 3rd party libraries and verify any calls within callbacks or after promises are still recording spans/metrics/etc

### PR DESCRIPTION
**Pull request by bizob2828**
## Description

Provides a new configuration stanza of `config.instrumentation.<library>.enabled` to disable every 3rd party library.  

* Use at your own risk. 
* Disabling instrumentation for a 3rd party library may affect instrumentation of libraries used after the disabled library. 

## How to Test

```sh
npm run versioned:internal major disabled-instrumentation
```

## Related Issues
Closes #2536

### CODE REVIEW

File: test/lib/custom-assertions.js [hunk 1]
```diff
@@ -80,6 +80,95 @@ function compareSegments(parent, segments) {
...
+            '" should have child "' +
+            sequenceItem +
+            '" in position ' +
+            childCount
+        )
+
+        // If the next expected item is not array, then check that the current
+        // child has no children
+        if (!Array.isArray(expected[i + 1])) {
+          // var children = child.children
```
Comment (mrickard) [inline test/lib/custom-assertions.js:139]: Should this commented-out line be removed?

File: lib/config/default.js [hunk 2]
```diff
@@ -1382,7 +1383,13 @@ defaultConfig.definition = () => ({
...
       }
     }
-  }
+  },
+  /**
+   * Stanza that contains all keys to disable 3rd party package instrumentation(i.e. mongodb, pg, redis, etc)
+   * Note: Disabling a given 3rd party library may affect the instrumentation of 3rd party libraries used after
+   * the disabled library.
+   */
+  instrumentation: pkgInstrumentation
```
Comment (jsumners-nr) [inline lib/config/default.js:1392]:
```suggestion
  /**
   * Stanza that contains all keys to disable 3rd party package instrumentation(i.e. mongodb, pg, redis, etc)
   * Note: Disabling a given 3rd party library may affect the instrumentation of 3rd party libraries used by
   * the disabled library.
   */
  instrumentation: pkgInstrumentation
```

File: test/versioned/disabled-instrumentation/package.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  "targets": [],
+  "version": "0.0.0",
+  "private": true,
+  "tests": [
+    {
+      "engines": {
+        "node": ">=18"
+      },
+      "dependencies": {
+        "express": "latest",
```
Comment (jsumners-nr) [inline test/versioned/disabled-instrumentation/package.json:12]: We should at least pin this one to v4 until we can figure out what is happening with v5 and the security agent.

File: test/lib/custom-assertions.js (hunk 1 above)
Comment (bizob2828) [reply test/lib/custom-assertions.js:139]: i can remove. i copied this function and made it non tap assertions.

File: lib/config/default.js (hunk 2 above)
Comment (bizob2828) [reply lib/config/default.js:1392]: i meant to say any instrumented libraries used after the disabled instrumented, it's not really by the disabled library

**Review by jsumners-nr — APPROVED** [review]

**Pull request merged**